File: app/src/main/java/forpdateam/ru/forpda/api/qms/models/QmsChatModel.java
package forpdateam.ru.forpda.api.qms.models;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by radiationx on 20.09.16.
 */

public class QmsChatModel {
    private int themeId = 0;
    private int userId = 0;
    private int showedMessIndex = 0;
    private String title;
    private String nick;
    private String avatarUrl;
    private String html;
    private List<QmsMessage> messages = new ArrayList<>();

    public void addMessage(QmsMessage item) {
        messages.add(item);
    }

    public List<QmsMessage> getMessages() {
        return messages;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getNick() {
        return nick;
    }

    public void setNick(String nick) {
        this.nick = nick;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public void setAvatarUrl(String avatarUrl) {
        this.avatarUrl = avatarUrl;
    }

    public int getThemeId() {
        return themeId;
    }

    public void setThemeId(int themeId) {
        this.themeId = themeId;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getHtml() {
        return html;
    }

    public void setHtml(String html) {
        this.html = html;
    }

    public int getShowedMessIndex() {
        return showedMessIndex;
    }

    public void setShowedMessIndex(int showedMessIndex) {
        this.showedMessIndex = showedMessIndex;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/entity/remote/qms/QmsChatModel.kt
package forpdateam.ru.forpda.entity.remote.qms

import forpdateam.ru.forpda.entity.DeferredData
import forpdateam.ru.forpda.entity.remote.others.user.ForumUser
import ru.radiationx.coretypes.QmsChatId

/**
 * Created by radiationx on 20.09.16.
 */

data class QmsChatModel(
    val id: QmsChatId,
    val user: ForumUser,
    val title: String,
    val messages: List<QmsMessage>,
    val showedMessIndex: Int,
    val html: DeferredData<String>?
) {

    companion object {
        const val NOT_CREATED = -1
    }
}
